modules/rpc/yarpc_options.go

```diff
@@ -32,28 +32,29 @@ const (
 )
 
 // WithInboundMiddleware adds custom YARPC inboundMiddleware to the module
-func WithInboundMiddleware(i ...middleware.UnaryInbound) modules.Option {
-	return func(mci *service.ModuleCreateInfo) error {
-		inboundMiddleware := inboundMiddlewareFromCreateInfo(*mci)
-		inboundMiddleware = append(inboundMiddleware, i...)
-		mci.Items[_interceptorKey] = inboundMiddleware
-
-		return nil
-	}
+func WithInboundMiddleware(i ...middleware.UnaryInbound) service.ModuleOption {
+	return service.WithModuleItem(_interceptorKey, func(existing interface{}) interface{} {
```

Comment:
I like this wrapping on write to module item.
maybe a similar wrapping for read as well?